### PR TITLE
JDK-8303354: addCertificatesToKeystore in KeystoreImpl.m needs CFRelease call in early potential CHECK_NULL return

### PR DESCRIPTION
**Pull request by MBaesken**
We have a (potential) early return in addCertificatesToKeystore in KeystoreImpl.m . This is implemented by the CHECK_NULL macro. However this missed a CFRelease call.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8303354](https://bugs.openjdk.org/browse/JDK-8303354): addCertificatesToKeystore in KeystoreImpl.m needs CFRelease call in early potential CHECK_NULL return

### Reviewers
 * [Christoph Langer](https://openjdk.org/census#clanger) (@RealCLanger - **Reviewer**) ⚠️ Review applies to [5508cdba](https://git.openjdk.org/jdk/pull/12788/files/5508cdba2163cae63c851a8861f932d5c724cef9)
 * [Martin Doerr](https://openjdk.org/census#mdoerr) (@TheRealMDoerr - **Reviewer**)
 * [Weijun Wang](https://openjdk.org/census#weijun) (@wangweij - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/12788/head:pull/12788` \
`$ git checkout pull/12788`

Update a local copy of the PR: \
`$ git checkout pull/12788` \
`$ git pull https://git.openjdk.org/jdk pull/12788/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 12788`

View PR using the GUI difftool: \
`$ git pr show -t 12788`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/12788.diff">https://git.openjdk.org/jdk/pull/12788.diff</a>

</details>

### CODE REVIEW

**Review by RealCLanger — APPROVED** [review]
Makes sense.

**Comment by MBaesken** [comment]
> Makes sense.

Hi Christoph, thanks for the review !  
May I have a second review ?

Thanks, Matthias

**Review by TheRealMDoerr — APPROVED** [review]
LGTM.

File: src/java.base/macosx/native/libosxsecurity/KeystoreImpl.m
Comment (MBaesken) [reply src/java.base/macosx/native/libosxsecurity/KeystoreImpl.m:431]: Hi Weijun, yes I think you are right , according to https://developer.apple.com/documentation/security/1515366-seckeychainsearchcreatefromattri   we have to call  CFRelease on keychainItemSearch

**Review by TheRealMDoerr — APPROVED** [review]
Thanks for the update!

File: src/java.base/macosx/native/libosxsecurity/KeystoreImpl.m
Comment (MBaesken) [reply src/java.base/macosx/native/libosxsecurity/KeystoreImpl.m:445]: > Thanks for the fix. I almost think we should invent a CHECK_NULL_GOTO_ERROUT macro, but your fix is also OK.

I think I  discussed a while back some extended CHECK_NULL_  - macros (I think it was for freeing before return) but it was not s well received.

**Comment by MBaesken** [comment]
Hi Martin and Weijun, thanks for the reviews !

**Pull request closed without merging**
